bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -0,0 +1,999 @@
+	ToolItem searchDownButton;
+	ToolItem searchAllButton;
+
+	Composite replaceContainer;
+	Composite replaceBarContainer;
+	Text replaceBar;
+	ToolBar replaceTools;
+	ToolItem replaceButton;
+	ToolItem replaceAllButton;
+
+	Color backgroundToUse;
+	Color normalTextForegroundColor;
+
+	IPreferenceChangeListener overlayDialogPreferenceListener = new IPreferenceChangeListener() {
+
+		@Override
+		public void preferenceChange(PreferenceChangeEvent event) {
+			if (event.getKey().equals("usefindreplaceoverlay")) { //$NON-NLS-1$
+				close();
+			}
+		}
+
+	};
+
+	public FindReplaceOverlay(Shell parent, IWorkbenchPart part, IFindReplaceTarget target,
+			FindReplaceAction parentAction) {
+		super(parent);
+		this.parentAction = parentAction;
+		createFindReplacer(target);
+
+		setShellStyle(SWT.MODELESS);
+		setBlockOnOpen(false);
+		targetPart = part;
+
+		IEclipsePreferences preferences = InstanceScope.INSTANCE.getNode("org.eclipse.ui.editors"); //$NON-NLS-1$
+		preferences.addPreferenceChangeListener(overlayDialogPreferenceListener);
+	}
+
+	@Override
+	protected boolean isResizable() {
+		return false;
+	}
+
+	private void createFindReplacer(IFindReplaceTarget target) {
+		findReplaceLogic = new FindReplaceLogic();
+		boolean isTargetEditable = false;
+		if (target != null) {
+			isTargetEditable = target.isEditable();
+		}
+		findReplaceLogic.updateTarget(target, isTargetEditable);
+		findReplaceLogic.activate(SearchOptions.INCREMENTAL);
+		findReplaceLogic.activate(SearchOptions.GLOBAL);
+		findReplaceLogic.activate(SearchOptions.WRAP);
+		findReplaceLogic.activate(SearchOptions.FORWARD);
+	}
+
+	KeyListener shortcuts = new KeyListener() {
+
+		private void performEnterAction(KeyEvent e) {
+			boolean isShiftPressed = (e.stateMask & SWT.SHIFT) != 0;
+			boolean isCtrlPressed = (e.stateMask & SWT.CTRL) != 0;
+			if (okayToUse(replaceBar) && replaceBar.isFocusControl()) {
+				if (isCtrlPressed) {
+					findReplaceLogic.performReplaceAll(getFindString(), getReplaceString(), getShell().getDisplay());
+					evaluateFindReplacerStatus();
+				} else {
+					performSingleReplace();
+					evaluateFindReplacerStatus();
+				}
+			} else {
+				if (isCtrlPressed) {
+					findReplaceLogic.performSelectAll(getFindString(), getShell().getDisplay());
+					evaluateFindReplacerStatus();
+				} else {
+					boolean oldForwardSearchSetting = findReplaceLogic.isActive(SearchOptions.FORWARD);
+					activateInFindReplacerIf(SearchOptions.FORWARD, !isShiftPressed);
+					findReplaceLogic.deactivate(SearchOptions.INCREMENTAL);
+					findReplaceLogic.performSearch(getFindString());
+					evaluateFindReplacerStatus();
+					activateInFindReplacerIf(SearchOptions.FORWARD, oldForwardSearchSetting);
+					findReplaceLogic.activate(SearchOptions.INCREMENTAL);
+				}
+			}
+		}
+
+		@Override
+		public void keyPressed(KeyEvent e) {
+			e.doit = false;
+			if ((e.stateMask & SWT.CTRL) != 0 && (e.keyCode == 'F' || e.keyCode == 'f')) {
+				parentAction.closeModernOverlay();
```

Comment:
Did you pick those method-names for a specific reason? To me it would make sense to make the names consistent with the names found in SWT-Listeners, eg

`addOverlayCloseListener`

Reply:
I don't believe such a listener is needed anymore. I now only perform this.close();
The reason I had added this is that previously, the FindReplaceAction kept track of whether the overlay was closed or opened. However, the Dialog now keeps track of that itself.